engines/e_zx.h

```diff
@@ -0,0 +1,57 @@
+#ifndef HEADER_SM4_H
+# define HEADER_SM4_H
```

Comment:
The name `HEADER_SM4_H` looks confusing...

Reply:
I have modified them according to your comments, thanks a lot.